icu4c/source/i18n/islamcal.cpp

```diff
@@ -877,7 +867,8 @@ int32_t IslamicUmalquraCalendar::handleGetMonthLength(int32_t extendedYear, int3
         }
         return length;
     }
-    return getUmalqura_MonthLength(extendedYear - UMALQURA_YEAR_START, month);
+    int32_t mask = (int32_t) (0x01 << (11 - month));    // set mask for bit corresponding to month
+    return 29 + (((UMALQURA_MONTHLENGTH[extendedYear - UMALQURA_YEAR_START] & mask) != 0) ? 1 : 0);
```

Comment:
I think this would probably be a little clearer as

```
return ((UMALQURA_MONTHLENGTH[extendedYear - UMALQURA_YEAR_START] & mask) != 0) ? 30 : 29;
```